src/test/util/blocktools.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2019 The Unit-e developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/util/blocktools.h>
+
+namespace blocktools {
+
+CBlockIndex *BlockIndexFake::MakeBlockIndex(const uint256 &hash, CBlockIndex *const prev) {
+  auto result = block_indexes.emplace(hash, CBlockIndex());
+  CBlockIndex *const index = &result.first->second;
+  index->phashBlock = &result.first->first;
+  index->nHeight = prev ? prev->nHeight + 1 : 0;
+  index->pprev = prev;
+  return index;
+}
+
+CBlockIndex *BlockIndexFake::Generate(const std::size_t count, const uint256 &starting_point) {
+  auto starting_block = block_indexes.find(starting_point);
+  std::size_t height = 0;
+  CBlockIndex *const starting_index = [&]() {
+    if (starting_block == block_indexes.end()) {
+      ++height;
+      return MakeBlockIndex(starting_point, nullptr);
+    } else {
+      return &starting_block->second;
+    }
+  }();
+  assert(starting_index);
```

Comment:
I think we tend to use BOOST_REQUIRE in unit tests (even in utilities of unit tests).

Reply:
Changed in d9ee56dbf059840357eb70fe33ddb0d92d971fbe

I was wondering about this myself when writing this code. I now changed but I am undecided on this one. `BOOST_REQUIRE` checks requirements of a specific test, this here really is about the correctness of the implementation and does not have anything to do per se with the test using it. In fact it can only go wrong if the implementation is changed as what is checked here are invariants relating to this very code itself.

Anyways, changed.

Reply:
I'll keep it like this. In some places this enabled me to do `BOOST_REQUIRE_MESSAGE` which is a bit nicer then the usual `assert(condition && "message")`.